src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1120,16 +1115,17 @@ namespace cryptonote
   void core::start_lokimq() {
       update_lmq_sns(); // Ensure we have SNs set for the current block before starting
 
+      if (m_service_node)
+        m_pulse_thread_id = m_lmq->add_tagged_thread("pulse");
+
+      m_lmq->start();
+
       if (m_service_node)
       {
-        m_blockchain_storage.hook_block_added(m_pulse_state);
-        lokimq::TaggedThreadID pulse_thread_id = m_lmq->add_tagged_thread("pulse");
-        m_lmq->start();
-        m_lmq->job([this]() { pulse::main(m_pulse_state, m_quorumnet_state, *this); }, pulse_thread_id);
-      }
-      else
-      {
-        m_lmq->start();
+        m_lmq->add_timer([this]() { pulse::main(m_quorumnet_state, *this); },
```

Comment:
This `add_timer` can go up with the `add_tagged_thread`; it supports being set up before start() is called (it won't trigger until after startup, of course).

Reply:
Updated